WireDomain/Tests/WireDomainTests/Synchronization/PushSupportedProtocolsUseCaseTests.swift

```diff
@@ -0,0 +1,362 @@
+        // Given
+
+        try await mock(allActiveMLSClients: false)
+        await mock(remoteSupportedProtocols: [.proteus])
+
+        mockSelfUserAPI.pushSupportedProtocols_MockMethod = { _ in }
+
+        let testCases: [(migrationState: Scaffolding.MigrationState, supportedProtocols: Set<WireAPI.MessageProtocol>)] = [
+            (migrationState: .disabled, supportedProtocols: [.proteus]),
+            (migrationState: .notStarted, supportedProtocols: [.proteus]),
+            (migrationState: .ongoing, supportedProtocols: [.proteus]),
+            (migrationState: .finalised, supportedProtocols: [.proteus])
+        ]
+
+        for testCase in testCases {
+            await mock(migrationState: testCase.migrationState)
+            // When
+            try await sut.invoke()
+            let pushedProtocols = try XCTUnwrap(mockSelfUserAPI.pushSupportedProtocols_Invocations.last)
+            // Then
+            XCTAssertEqual(testCase.supportedProtocols, pushedProtocols)
+        }
+    }
+
+    func test_CalculateSupportedProtocols_NotAllActiveMLSClients_RemoteProteusAndMLS() async throws {
+        // Given
+
+        try await mock(allActiveMLSClients: false)
+        await mock(remoteSupportedProtocols: [.proteus, .mls])
+
+        mockSelfUserAPI.pushSupportedProtocols_MockMethod = { _ in }
+
+        let testCases: [(migrationState: Scaffolding.MigrationState, supportedProtocols: Set<WireAPI.MessageProtocol>)] = [
+            (migrationState: .disabled, supportedProtocols: [.proteus]),
+            (migrationState: .notStarted, supportedProtocols: [.proteus]),
+            (migrationState: .ongoing, supportedProtocols: [.proteus]),
+            (migrationState: .finalised, supportedProtocols: [.proteus, .mls])
+        ]
+
+        for testCase in testCases {
+            await mock(migrationState: testCase.migrationState)
+            // When
+            try await sut.invoke()
+            let pushedProtocols = try XCTUnwrap(mockSelfUserAPI.pushSupportedProtocols_Invocations.last)
+            // Then
+            XCTAssertEqual(testCase.supportedProtocols, pushedProtocols)
+        }
+    }
+
+    func test_CalculateSupportedProtocols_NotAllActiveMLSClients_RemoteMLS() async throws {
+        // Given
+
+        try await mock(allActiveMLSClients: false)
+        await mock(remoteSupportedProtocols: [.mls])
+
+        mockSelfUserAPI.pushSupportedProtocols_MockMethod = { _ in }
+
+        let testCases: [(migrationState: Scaffolding.MigrationState, supportedProtocols: Set<WireAPI.MessageProtocol>)] = [
+            (migrationState: .disabled, supportedProtocols: [.mls]),
+            (migrationState: .notStarted, supportedProtocols: [.proteus]),
+            (migrationState: .ongoing, supportedProtocols: [.proteus]),
+            (migrationState: .finalised, supportedProtocols: [.mls])
+        ]
+
+        for testCase in testCases {
+            await mock(migrationState: testCase.migrationState)
```

Comment:
nit-pick: are we really mocking or is it more setup?
```suggestion
            await setup(migrationState: testCase.migrationState)
```
same for all mock methods

Reply:
fixed thx